File: Frontend/brain/e366d5b6-c3d1-4a1c-b34d-02fbacb142f6/scratch/check_tags.js
const fs = require('fs');

function checkBalance(filePath) {
    let content = fs.readFileSync(filePath, 'utf8');
    
    // Remove comments
    content = content.replace(/\{\/\*.*?\*\/\}/gs, '');
    content = content.replace(/\/\/.*$/gm, '');
    
    const tagRegex = /<(/?)([a-zA-Z0-9\.]+)/g;
    let match;
    let stack = [];
    const selfClosing = ['input', 'img', 'br', 'hr', 'Bell', 'HelpCircle', 'Menu', 'ArrowLeft', 'Search', 'SlidersHorizontal', 'Loader2', 'Star', 'X'];

    while ((match = tagRegex.exec(content)) !== null) {
        const isClosing = match[1] === '/';
        const tagName = match[2];

        if (selfClosing.includes(tagName)) continue;

        if (isClosing) {
            if (stack.length === 0) {
                console.log(`Extra closing tag: ${tagName}`);
            } else {
                const top = stack.pop();
                if (top !== tagName) {
                    console.log(`Mismatched tags: opened ${top}, closing ${tagName}`);
                }
            }
        } else {
            // Check if it's self-closing like <img />
            const restOfTag = content.substring(match.index, content.indexOf('>', match.index) + 1);
            if (restOfTag.endsWith('/>')) continue;

            stack.push(tagName);
        }
    }
    
    if (stack.length > 0) {
        console.log(`Unclosed tags: ${stack.join(', ')}`);
    } else {
        console.log("All tags balanced");
    }
}

checkBalance('c:\\Users\\aditi\\OneDrive\\Desktop\\company project\\iggymet main\\Frontend\\src\\modules\Food\\pages\\restaurant\\Feedback.jsx');

File: Frontend/brain/e366d5b6-c3d1-4a1c-b34d-02fbacb142f6/scratch/check_tags.py
import re

def check_balance(file_path):
    with open(file_path, 'r', encoding='utf-8') as f:
        content = f.read()
    
    # Remove comments
    content = re.sub(r'\{/\*.*?\*/\}', '', content, flags=re.DOTALL)
    content = re.sub(r'//.*', '', content)
    
    tags = re.findall(r'<(/?)([a-zA-Z0-9\.]+)', content)
    stack = []
    for is_closing, tag_name in tags:
        if tag_name in ['input', 'img', 'br', 'hr', 'Bell', 'HelpCircle', 'Menu', 'ArrowLeft', 'Search', 'SlidersHorizontal', 'Loader2', 'Star', 'X']:
            continue
        if is_closing:
            if not stack:
                print(f"Extra closing tag: {tag_name}")
            else:
                top = stack.pop()
                if top != tag_name:
                    print(f"Mismatched tags: opened {top}, closing {tag_name}")
        else:
            stack.append(tag_name)
    
    if stack:
        print(f"Unclosed tags: {stack}")
    else:
        print("All tags balanced")

check_balance(r'c:\Users\aditi\OneDrive\Desktop\company project\iggymet main\Frontend\src\modules\Food\pages\restaurant\Feedback.jsx')
